applications/new_schema.py

```diff
@@ -84,9 +85,7 @@ def resolve_harbor_choices(self, info, **kwargs):
         return None
 
     @classmethod
-    @login_required
-    @superuser_required
-    # TODO: Should check if the user has permissions to access this specific object
+    @view_permission_required(BerthApplication)
```

Comment:
what do you think? should we list all the types that `BerthApplicationNode` has relations to, like `HarborChoice, CustomerProfile, Company, Boat, Lease`? or is it safe / sane enough to only check for the top level model, i.e. `BerthApplication`? they all are within the same business logic category in the `User roles` table, but technically different models. I am not sure. 🤔

Reply:
Good catch, I think we should check for the non-public models, since the gruop permissions could _theoretically_ change, so better safe than sorry.

Reply:
Ok, let's see how it will look if we check for all relevant models. 👍